File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        long long ones = 0, left = 0, right = (int) s.size() - 1, answer = 0;

        while (~right and s[right] == '1') --right;

        while (left <= right) {
            if (s[left] == '1')
                while (left <= right and s[left] == '1') ++left, ones++;
            else
                answer += ones, ++left;
        }
        return answer;
    }
};

File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumSteps = function(s) {
    let left = 0, right = s.length - 1, answer = 0, ones = 0;
    while(right != -1 && s[right] == '1') right -= 1;
    
        while (left <= right) {
            if (s[left] == '1')
                while (left <= right && s[left] == '1') {
                    left += 1;
                    ones += 1;
                }
            else{
                answer += ones;
                left += 1;
            }
        }
        return answer;

};